### ISSUE TITLE
refactor custom element definitions being shared

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/wcc#24]
## Type of Change

- Other (please clarify below)

## Summary
As encountered in #21 , it was determined through mocha testing that having `definitions` outside of any function, would effectively act as a global variable between runs, or otherwise cause caching issues in NodeJS between `renderToString` calls.

## Details
I had always wanted a better way to code this initially as `definitions` is updated / read globally by all those functions, so is effectively global state, which is not idea.  And as revealed in that PR and verified in https://github.com/mochajs/mocha/issues/1637, this causes odd behaviors.

So basically we need to get rid of `let definitions` as a shared variable, and still somehow make it all work.  Also open to using better data structures if it fits, like a `Map` or `Set`.

Also seeing an issue in Greenwood [when using `metadata` for hydration](https://github.com/ProjectEvergreen/greenwood/pull/943/files#diff-610bc20aece54908931ddf5cf549d236a44e8d5ce78c314abf02fde24ab32c97R30) it was [breaking GitHub Actions on Windows](https://github.com/ProjectEvergreen/greenwood/runs/6952039062?check_suite_focus=true).  😬
